test/interactions/clickInteractionTests.ts

```diff
@@ -1,158 +1,208 @@
-        assert.isTrue(callbackCalled, "callback called on clicking Component without moving mouse");
-        assert.deepEqual(lastPoint, { x: SVG_WIDTH / 2, y: SVG_HEIGHT / 2 }, "was passed correct point");
-
-        callbackCalled = false;
-        lastPoint = null;
-        TestMethods.triggerFakeMouseEvent("mousedown", component.content(), SVG_WIDTH / 2, SVG_HEIGHT / 2);
-        TestMethods.triggerFakeMouseEvent("mouseup", component.content(), SVG_WIDTH / 4, SVG_HEIGHT / 4);
-        assert.isTrue(callbackCalled, "callback called on clicking and releasing inside the Component");
-        assert.deepEqual(lastPoint, { x: SVG_WIDTH / 4, y: SVG_HEIGHT / 4 }, "was passed mouseup point");
-
-        callbackCalled = false;
-        TestMethods.triggerFakeMouseEvent("mousedown", component.content(), SVG_WIDTH / 2, SVG_HEIGHT / 2);
-        TestMethods.triggerFakeMouseEvent("mouseup", component.content(), SVG_WIDTH * 2, SVG_HEIGHT * 2);
-        assert.isFalse(callbackCalled, "callback not called if mouse is released outside Component");
-
-        TestMethods.triggerFakeMouseEvent("mousedown", component.content(), SVG_WIDTH * 2, SVG_HEIGHT * 2);
-        TestMethods.triggerFakeMouseEvent("mouseup", component.content(), SVG_WIDTH / 2, SVG_HEIGHT / 2);
-        assert.isFalse(callbackCalled, "callback not called if click was started outside Component");
-
-        TestMethods.triggerFakeMouseEvent("mousedown", component.content(), SVG_WIDTH / 2, SVG_HEIGHT / 2);
-        TestMethods.triggerFakeMouseEvent("mousemove", component.content(), SVG_WIDTH * 2, SVG_HEIGHT * 2);
-        TestMethods.triggerFakeMouseEvent("mouseup", component.content(), SVG_WIDTH / 2, SVG_HEIGHT / 2);
-        assert.isTrue(callbackCalled, "callback still called if the mouse is moved out then back inside the Component before releasing");
-
-        svg.remove();
-      });
+  afterEach(() => {
+    svg.remove();
+  });
 
-      it("treats touch events same as click events for callbacks registered with onClick()", () => {
-        let callbackCalled = false;
-        let lastPoint: Plottable.Point;
-        let callback = function(point: Plottable.Point) {
-          callbackCalled = true;
-          lastPoint = point;
-        };
-        clickInteraction.onClick(callback);
+  describe("onClick/offClick", () => {
+    describe("registration", () => {
+      const testPoint = {x: 0, y: 0};
 
-        TestMethods.triggerFakeTouchEvent("touchstart", component.content(), [{x: SVG_WIDTH / 2, y: SVG_HEIGHT / 2}]);
-        TestMethods.triggerFakeTouchEvent("touchend", component.content(), [{x: SVG_WIDTH / 2, y: SVG_HEIGHT / 2}]);
-        assert.isTrue(callbackCalled, "callback called on clicking Component without moving mouse");
-        assert.deepEqual(lastPoint, { x: SVG_WIDTH / 2, y: SVG_HEIGHT / 2 }, "was passed correct point");
+      it("registers callback using onClick", () => {
+        const callback = new TestClickCallback();
```

Comment:
Looks like every test basically wants the same type of callback. We could put it in the `beforeEach`, but some test cases need two of them.